kube-client/src/client/mod.rs

```diff
@@ -239,14 +239,44 @@ impl Client {
     }
 
     /// Perform a raw HTTP request against the API and get back the response
-    /// as a stream of bytes
+    /// as a stream of strings where each chunk represents a log line.
     pub async fn request_text_stream(
         &self,
         request: Request<Vec<u8>>,
-    ) -> Result<impl Stream<Item = Result<Bytes>>> {
+    ) -> Result<impl Stream<Item = Result<String>>> {
         let res = self.send(request.map(Body::from)).await?;
-        // trace!("Status = {:?} for {}", res.status(), res.url());
-        Ok(res.into_body().map_err(Error::HyperError))
+        let frames = FramedRead::new(
+            StreamReader::new(res.into_body().map_err(|e| {
+                // Client timeout. This will be ignored.
+                if e.is_timeout() {
+                    return std::io::Error::new(std::io::ErrorKind::TimedOut, e);
+                }
+                std::io::Error::new(std::io::ErrorKind::Other, e)
+            })),
+            LinesCodec::new(),
+        );
+        Ok(frames.filter_map(|res| async {
+            match res {
+                Ok(val) => {
+                    Some(Ok(val))
+                },
+
+                Err(LinesCodecError::Io(e)) => match e.kind() {
+                    // Client timeout
+                    std::io::ErrorKind::TimedOut => {
+                        tracing::warn!("timeout in poll: {}", e); // our client timeout
+                        None
+                    }
```

Comment:
In `request_events`, timeouts are treated as errors and bubbled up, here it closes the stream. Is there a good reason for this?

Reply:
i don't think `request_events` bubbles up the timeout error:
https://github.com/kube-rs/kube/blob/f792fd22d7568496447012ace4fe2f13d0e59def/kube-client/src/client/mod.rs#L324-L327

or am i misunderstanding something?

Reply:
oh, you are right. i quickly skimmed here https://github.com/kube-rs/kube/blob/f792fd22d7568496447012ace4fe2f13d0e59def/kube-client/src/client/mod.rs#L290-L292
then failed to see that it was matched onlater.